packages/ember-glimmer/tests/integration/input-test.js

```diff
@@ -0,0 +1,155 @@
+import { RenderingTest, moduleFor } from '../utils/test-case';
+import { set } from 'ember-metal/property_set';
+
+moduleFor('Input element tests', class extends RenderingTest {
+  runAttributeTest(attributeName, values) {
+    let template = `<input ${attributeName}={{value}}>`;
+    this.render(template, { value: values[0] });
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} is set on initial render`);
+
+    this.runTask(() => this.rerender());
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} is set on noop rerender`);
+
+    this.setComponentValue(values[1]);
+    this.assertAttributeHasValue(attributeName, values[1], `${attributeName} is set on rerender`);
+
+    this.setComponentValue(values[0]);
+    this.assertAttributeHasValue(attributeName, values[0], `${attributeName} can be set back to the initial value`);
+  }
+
+  runPropertyTest(propertyName, values) {
+    let attributeName = propertyName;
+    let template = `<input ${attributeName}={{value}}>`;
+    this.render(template, { value: values[0] });
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} is set on initial render`);
+
+    this.runTask(() => this.rerender());
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} is set on noop rerender`);
+
+    this.setComponentValue(values[1]);
+    this.assertPropertyHasValue(propertyName, values[1], `${propertyName} is set on rerender`);
+
+    this.setComponentValue(values[0]);
+    this.assertPropertyHasValue(propertyName, values[0], `${propertyName} can be set back to the initial value`);
+  }
+
+  ['@test input disabled attribute']() {
+    this.runPropertyTest('disabled', [false, true]);
+  }
+
+  ['@test input value attribute']() {
+    this.runPropertyTest('value', ['foo', 'bar']);
+  }
+
+  ['@htmlbars input placeholder attribute']() {
```

Comment:
I'm actually not sure, in IE9 this fails and I think it has to do with attributes vs. properties.